optimize/minimize.go

```diff
@@ -480,7 +480,11 @@ func evaluate(p *Problem, loc *Location, op Operation, x []float64) {
 		loc.Gradient = p.Grad(loc.Gradient, x)
 	}
 	if op&HessEvaluation != 0 {
-		loc.Hessian = p.Hess(loc.Hessian, x)
+		// Make sure we have a destination in which to place the Hessian.
```

Comment:
This is here because it makes the initialisation vastly more convenient at the cost of some simple pixie dust. There is no documentation for this, though it could be added to `Location`, saying that `Hessian` will be allocated if needed.

Reply:
SGTM.